CMakeLists.txt

```diff
@@ -158,7 +158,7 @@ if (FL_BUILD_CORE)
 
   # FIXME: libraries are required to build core tests - https://git.io/JL955
   if (FL_BUILD_LIBRARIES OR (FL_BUILD_CORE AND FL_BUILD_TESTS))
-    target_link_libraries(flashlight PUBLIC fl-libraries)
+    target_link_libraries(flashlight PUBLIC fl_lib_common)
```

Comment:
This might be fixed in https://github.com/padentomasello/flashlight/pull/17/files (I will need to check), but we should only build `fl_lib_common` if this above condition is true, else it isn't required for the core

Reply:
This should be the case since we have the same condition gating building libraries here: https://github.com/padentomasello/flashlight/blob/dev/paden/lib-refactor/CMakeLists.txt#L118. 

Now that I'm looking at this again, I realize libraries are all or nothing -- I'm assuming we probably want an FL_BUILD_LIB_COMMON yeah?

Reply:
Ah I see you have similar comments in later PRs. I think we are on the same page. I will add these gates.